_posts/2023-08-22-using-plurimath.adoc

```diff
@@ -0,0 +1,282 @@
+With just a few lines of code, you can harness the power of *Plurimath* to ensure consistency and compatibility in mathematical presentations across different platforms and applications.
+
+Now let's discuss *how to use Plurimath?*.
+
+```ruby
+  formula = Plurimath::Math.parse(string, :type)
+```
+
+You will have to provide input text replacing `string`, and the `type` field should specify the type of the provided text, choosing from one of the supported languages.
+
+The above example will generate a Formula object, something like this
+
+```ruby
+  Plurimath::Math::Formula.new([
+    ... # Others objects
+  ])
+```
+
+Now next step would be conversion(since you parsed your string into *Plurimath Formula*) to your desired output language. For conversion all you have to do is call method name containing language's name with leading `to_`.
+
+```ruby
+  formula.to_<language-name>
+```
+or you can do it directly,
+```ruby
+  Plurimath::Math.parse(string, :type).to_<language-name>
+```
+
+NOTE: Some methods support explicit optional argument and some don't.
+
+Below sections will be explaining each supported languages conversion and parsing with an example
+
+=== 1.2 Use of Plurimath, Explained
+
+Below sections will help understand using *Plurimath* with realtime example.
+We have an *AsciiMath* equation `sum_(i=1)^n i^3`, we will be using this equation in all the sections.
+
+==== 1.2.1 AsciiMath to Formula
+
+```ruby
+  input_string = 'sum_(i=1)^n i^3'
+  formula = Plurimath::Math.parse(input_string, :asciimath)
+```
+Parsing the equation will generate following formula.
+```ruby
+  Plurimath::Math::Formula.new([
+    Plurimath::Math::Function::Sum.new(
+      Plurimath::Math::Formula.new([
+        Plurimath::Math::Symbol.new("i"),
+        Plurimath::Math::Symbol.new("="),
+        Plurimath::Math::Number.new("1")
+      ]),
+      Plurimath::Math::Symbol.new("n"),
+      Plurimath::Math::Function::Power.new(
+        Plurimath::Math::Symbol.new("i"),
+        Plurimath::Math::Number.new("3")
+      )
+    )
+  ])
+```
+
+==== 1.2.2 Formula to AsciiMath
+
+Following example follows the same steps we have before for *AsciiMath* as well.
+
+```ruby
+  input_string = <<~MATHML
+    <math xmlns="http://www.w3.org/1998/Math/MathML" display="block">
+      <mstyle displaystyle="true">
+        <mrow>
+          <munderover>
+            <mo>&#x2211;</mo>
+            <mrow>
+              <mi>i</mi>
+              <mo>=</mo>
+              <mn>1</mn>
+            </mrow>
+            <mi>n</mi>
+          </munderover>
+          <msup>
+            <mi>i</mi>
+            <mn>3</mn>
+          </msup>
+        </mrow>
+      </mstyle>
+    </math>
+  MATHML
+  formula = Plurimath::Math.parse(input_string, :mathml)
+  formula.to_asciimath # >> "sum_(i=1)^n i^3"
+```
+
+NOTE: Heredoc is used above for better readability, but you're also welcome to use quotes to create strings if you prefer.
```

Comment:
is this really needed? We assume readers know Ruby, which means they know that heredocs are strings.